From d3a69e267774fd515730cace8683ad04f95f0daa Mon Sep 17 00:00:00 2001
From: Ryan Zhang <ryan.zhang@nxp.com>
Date: Sat, 11 Oct 2025 20:56:43 +0800
Subject: [PATCH 5/6] MGS-8467 [#nxp] adapt unique_ptr to free pointer in
 ~swapchain()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

m_wsi_allocator has been managed by unique_ptr in latest wsi,
so we shouldn't free it manually in ~swapchain().

fix the build issue:
vulkan-wsi-layer/wsi/wayland/swapchain.cpp:81:23: error: cannot convert
‘util::unique_ptr<wsi::swapchain_wsialloc_allocator>’ {aka ‘std::unique_ptr
<wsi::swapchain_wsialloc_allocator, util::deleter<wsi::swapchain_wsialloc
_allocator> >’} to ‘wsialloc_allocator*’  81 | wsialloc_delete(m_wsi_allocator);

Upstream-Status: Inappropriate [i.MX-specific]
Signed-off-by: Ryan Zhang <ryan.zhang@nxp.com>
---
 wsi/wayland/swapchain.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/wsi/wayland/swapchain.cpp b/wsi/wayland/swapchain.cpp
index 9a11515..ee6ddcb 100644
--- a/wsi/wayland/swapchain.cpp
+++ b/wsi/wayland/swapchain.cpp
@@ -76,11 +76,6 @@ swapchain::~swapchain()
 {
    teardown();
 
-   if (m_wsi_allocator != nullptr)
-   {
-      wsialloc_delete(m_wsi_allocator);
-   }
-   m_wsi_allocator = nullptr;
    if (wlc.opaque_region)
    {
       wl_compositor_destroy(wlc.wl_compositor);
-- 
2.34.1

